Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -4970,6 +4970,24 @@ bool RenderBox::establishesIndependentFormattingContext() const
     return isGridItem() || RenderElement::establishesIndependentFormattingContext();
 }
 
+bool RenderBox::createsBlockFormattingContext() const
+{
+    const auto& boxStyle = style();
+    if (!boxStyle.isDisplayBlockLevel())
+        return false;
+    auto isBlockWithOverFlowOtherThanVisibleAndClip = [&]() {
+        auto boxOverflowX = boxStyle.overflowX();
```

Comment:
should this use effectiveOverflow* instead?

Reply:
As far as I can tell there is no reason that it should not. Went ahead and changed it to use it